govcd/nsxv_dhcprelay_test.go

```diff
@@ -0,0 +1,103 @@
+// +build nsxv functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_NsxvDhcpRelay tests out Edge gateway DHCP relay configuration settings. It does the following:
+// 1. Creates an IP set to ensure it can be used in DHCP relay configuration settings.
+// 2. Creates a DHCP relay configuration with multiple objects and checks. Adds a task for cleanup.
+// Note. Because DHCP relay configuration is not an object, but a setting of edge gateway - there is
+// nothing to delete
+// 3. Resets the DHCP relay configuration and checks that it was removed
+// 4. Creates another DHCP relay with different configuration (using only IP sets) and manually
+// specifying IP address on edge gateway interface (vNic)
+// 6. Resets the DHCP relay configuration and checks that it was removed
+func (vcd *TestVCD) Test_NsxvDhcpRelay(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip("Skipping test because no edge gateway given")
+	}
+
+	edge, err := vcd.vdc.GetEdgeGatewayByName(vcd.config.VCD.EdgeGateway, false)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
+
+	// Setup IP set for testing purposes and add it to cleanup list
+	createdIpSet, err := testCreateIpSet("dhcp-relay-test", vcd.vdc)
+	check.Assert(err, IsNil)
+	parentEntity := vcd.org.Org.Name + "|" + vcd.vdc.Vdc.Name
+	AddToCleanupList(createdIpSet.Name, "ipSet", parentEntity, check.TestName())
+
+	// Lookup vNic index for our org network
+	vNicIndex, _, err := edge.GetAnyVnicIndexByNetworkName(vcd.config.VCD.Network.Net1)
+	check.Assert(err, IsNil)
+
+	dhcpRelayConfig := &types.EdgeDhcpRelay{
+		RelayServer: &types.EdgeDhcpRelayServer{
+			IpAddress:        []string{"1.1.1.1"},
+			Fqdns:            []string{"servergroups.domainname.com", "servergroups.otherdomainname.com"},
+			GroupingObjectId: []string{createdIpSet.ID},
+		},
+		RelayAgents: &types.EdgeDhcpRelayAgents{
+			Agents: []types.EdgeDhcpRelayAgent{
+				types.EdgeDhcpRelayAgent{
+					VnicIndex: vNicIndex,
+				},
+			},
+		},
+	}
+
+	createdRelayConfig, err := edge.UpdateDhcpRelay(dhcpRelayConfig)
+	check.Assert(err, IsNil)
+
+	parentEntity = vcd.org.Org.Name + "|" + vcd.vdc.Vdc.Name + "|" + vcd.config.VCD.EdgeGateway
+	PrependToCleanupList(check.TestName(), "dhcpRelayConfig", parentEntity, check.TestName())
```

Comment:
Please add a comment about using `PrependToCleanupList` instead of Add, for future reference

Reply:
Added